compiler/src/iree/compiler/Dialect/Flow/IR/FlowOpFolders.cpp

```diff
@@ -1192,11 +1192,35 @@ struct ElideRedundantTransfer : public OpRewritePattern<TensorTransferOp> {
   }
 };
 
+// Attempts to identify trivial case of chained transfer ops (A -> B -> C) and rewrite it as (A -> C)
+// Writes it as A -> B and A -> C relying on dead code elimination to remove the unused A -> B transfer.
+struct ElideChainedTransfer : public OpRewritePattern<TensorTransferOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(TensorTransferOp currTransferOp,
```

Comment:
style: avoid abbreviations in naming unless canonical ("curr" should be "current", "prev" is ok, but if you have "current" then "previous" is clearer)

"current" and "previous" aren't quite right, though, as it's really "source" and "target"